File: 백준/Silver/2670. 연속부분최대곱/연속부분최대곱.java


import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {
	// Main
	public static void main(String[] args) throws IOException {

		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

		int N = Integer.parseInt(br.readLine());

		float[] arr = new float[N];
		for (int i = 0; i < arr.length; i++) {
			arr[i] = Float.parseFloat(br.readLine());
		}
		float max = 0;
		for (int i = 0; i < N; i++) {
			float mul = 1;
			for (int j = i; j < N; j++) {
				mul *= arr[j];
				max = Math.max(max, mul);
//				System.out.println(mul);
			}
		}
		System.out.println(String.format("%.3f", (float) Math.round(max * 1000) / 1000));
	}
}

File: 백준/Silver/2670. 연속부분최대곱/연속부분최대곱.py
# ans = 0
# n = int(input())
# list = []
# for n in range(n):
#     list.append(float(input()))
# print(list)
# for i in range(n):
#     mul = 1
#     for j in range(i,n):
#         mul *= list[j]
#         ans= max(ans, mul)
# print("{0:.3f}".format(round(ans,3)))

ans = 0
n = int(input())
list = []
for n in range(n):
    list.append(float(input()))
dp = [0]* n
dp[0] = list[0]
for i in range(1,n):
    dp[i] = max(list[i], list[i]*dp[i-1])
# print(dp)
print("{0:.3f}".format(round(max(dp),3)))